osu.Game/Overlays/Settings/Sections/SkinSection.cs

```diff
@@ -59,15 +65,32 @@ private void load(OsuConfigManager config, SkinManager skins)
 
             config.BindWith(OsuSetting.Skin, configBindable);
 
+            usableSkins = skins.GetAllUsableSkins().ToArray();
```

Comment:
I don't know, I kept the previous code for this which converted the list to an array too, even though it accepts any Enumerable

Reply:
The `ToArray` can definitely be removed here.

Reply:
oh I see why it is there now. The itemRemoved/itemAdded callbacks call where/append and convert that to an array to avoid huge LINQ objects like `original source -> where -> where -> where -> where -> append -> where` etc.

I believe a linked list would be a good replacement to avoid new allocations and GC triggers on every added/removed skin

Reply:
In those cases it is understandable, but `GetAllUsableSkins` already returns a list, so it is not needed on this specific line.